File: server/src/main/java/com/github/mtgaccountant/server/service/SearchService.java
package com.github.mtgaccountant.server.service;

import java.util.List;

import org.springframework.http.ResponseEntity;

import com.github.mtgaccountant.server.models.ClientSearch;
import com.github.mtgaccountant.server.wrapper.CardSearchWrapper;

public interface SearchService {
    ResponseEntity<List<CardSearchWrapper>> searchCards(String name, String[] rarities, String[] setTypes,
            Integer minPrice,
            Integer maxPrice, String[] sets, String[] colors, String owned);
}

File: client/src/app/services/search.service.ts
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';

@Injectable({
    providedIn: 'root',
})
export class SearchService {
    url = environment.apiUrl;

    constructor(private httpClient: HttpClient) {}

    searchCards(searchData: any): Observable<any> {
        if (searchData.minPrice == null) {
            searchData.minPrice = 0;
        }

        if (searchData.maxPrice == null) {
            searchData.maxPrice = 0;
        }

        return this.httpClient.get(
            this.url +
                `/api/search/cards?name=${searchData.name}&rarities=${searchData.rarities}&setTypes=${searchData.setTypes}&minPrice=${searchData.minPrice}&maxPrice=${searchData.maxPrice}&sets=${searchData.sets}&colors=${searchData.colors}&owned=${searchData.owned}`,
            {
                headers: new HttpHeaders().set(
                    'Content-Type',
                    'application/json'
                ),
            }
        );
    }
}
